### PR TITLE
Fix matview access method change issue

### PR DESCRIPTION
**Pull request by agedemenli**
DESCRIPTION: Fixes matview access method change issue

fixes: #4707

### CODE REVIEW

File: src/backend/distributed/commands/alter_table.c [hunk 1]
```diff
@@ -1517,6 +1522,41 @@ CreateWorkerChangeSequenceDependencyCommand(char *sequenceSchemaName, char *sequ
 }
 
 
+/*
+ * GetUsingClauseWithAccessMethodIfExists generates and returns "USING ..." string
+ * if there's an access method set to the view with the given oid.
+ */
+static char *
+GetUsingClauseWithAccessMethodIfExists(Oid viewOid)
```
Comment (halilozanakgul) [inline src/backend/distributed/commands/alter_table.c:1530]: I'm not really happy with this function name.
Can you change it to something like `CreateUsingAccessMethodClauseForMatView` or something similar, if it makes sense?

File: src/backend/distributed/commands/alter_table.c [hunk 2]
```diff
@@ -1517,6 +1522,41 @@ CreateWorkerChangeSequenceDependencyCommand(char *sequenceSchemaName, char *sequ
...
+	if (relation == NULL)
+	{
+		ereport(ERROR, (errmsg("cannot complete operation "
+							   "because no such view exists")));
+	}
+
+	Oid accessMethodOid = relation->rd_rel->relam;
+	if (OidIsValid(accessMethodOid))
+	{
+		HeapTuple amTuple = SearchSysCache1(AMOID,
```
Comment (halilozanakgul) [inline src/backend/distributed/commands/alter_table.c:1543]: I think you can use `get_am_name` function here.

**Review by halilozanakgul — CHANGES_REQUESTED** [review]

File: src/backend/distributed/commands/alter_table.c (hunk 1 above)
Comment (agedemenli) [reply src/backend/distributed/commands/alter_table.c:1530]: I want to explicitly say that there might be no access method set for the given view. So how about `CreateUsingAccessMethodClauseForViewIfExists` ?

File: src/backend/distributed/commands/alter_table.c (hunk 2 above)
Comment (agedemenli) [reply src/backend/distributed/commands/alter_table.c:1543]: Will check that out

File: src/backend/distributed/commands/alter_table.c (hunk 1 above)
Comment (halilozanakgul) [reply src/backend/distributed/commands/alter_table.c:1530]: `CreateUsingAccessMethodClauseForMatViewIfExists` ?

Comment (agedemenli) [reply src/backend/distributed/commands/alter_table.c]: Deal :)

File: src/backend/distributed/commands/alter_table.c [hunk 3]
```diff
@@ -1517,6 +1524,33 @@ CreateWorkerChangeSequenceDependencyCommand(char *sequenceSchemaName, char *sequ
...
+	if (relation == NULL)
+	{
+		ereport(ERROR, (errmsg("cannot complete operation "
+							   "because no such view exists")));
+	}
+
+	Oid accessMethodOid = relation->rd_rel->relam;
+	if (OidIsValid(accessMethodOid))
+	{
+		char *accessMethodName = get_am_name(accessMethodOid);
```
Comment (halilozanakgul) [inline src/backend/distributed/commands/alter_table.c:1545]: I'm not sure if it is certainly necessary but I think we can check if we actually received a result from this function.

File: src/backend/distributed/commands/alter_table.c [hunk 4]
```diff
@@ -1126,10 +1128,15 @@ GetViewCreationCommandsOfTable(Oid relationId)
 		char *qualifiedViewName = quote_qualified_identifier(schemaName, viewName);
 		bool isMatView = get_rel_relkind(viewOid) == RELKIND_MATVIEW;
 
+		/* here we need to get the access method of the view to recreate it */
+		char *usingAccessMethod = CreateUsingAccessMethodClauseForMatViewIfExists(
+			viewOid);
+
 		appendStringInfo(query,
-						 "CREATE %s VIEW %s AS %s",
+						 "CREATE %s VIEW %s %s AS %s",
```
Comment (onderkalaci) [inline src/backend/distributed/commands/alter_table.c:1136]: it feels a little unexpected to append to the string even of `usingAccessMethod == NULL`. Can't we have a simple if/else?

Comment (agedemenli) [reply src/backend/distributed/commands/alter_table.c:1136]: It's never `NULL`. Might be empty string. We can check if it's empty or not, if that makes sense?

**Review by onderkalaci — APPROVED** [review]

**Pull request merged**